app/workers/argo_sync_worker.rb

```diff
@@ -0,0 +1,81 @@
+class ArgoSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: false
+
+  def perform(helm_infrastructure_id)
+    infrastructure = HelmInfrastructure.find(helm_infrastructure_id)
+
+    if infrastructure.blank? || !infrastructure.active?
+      render json: {
+        success: false,
+        errors: ["Infrastructure not found"],
+        code: 404
+      }, status: :not_found and return
```

Comment:
the perform function is async, can we render smth like this? the json couldn't be seen anywhere is it?

Reply:
updated